editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -3766,7 +3766,7 @@ void CanvasItemEditor::_draw_invisible_nodes_positions(Node *p_node, const Trans
 		_draw_invisible_nodes_positions(p_node->get_child(i), parent_xform, canvas_xform);
 	}
 
-	if (show_position_gizmos && ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci))) {
+	if ((!(hide_subscene_gizmos && !p_node->get_scene_file_path().is_empty())) && show_position_gizmos && ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci))) {
```

Comment:
```suggestion
	const bool is_hidden_subscene_gizmo = hide_subscene_gizmos && !p_node->get_scene_file_path().is_empty();
	if (!is_hidden_subscene_gizmo && show_position_gizmos && ci && !ci->_edit_use_rect() && (!editor_selection->is_selected(ci) || _is_node_locked(ci))) {
```
Instead of repeating it every time.

Reply:
Variable created (twice for two methods)